packages/babel-plugin-transform-typescript/src/index.ts

```diff
@@ -307,24 +328,33 @@ export default declare((api, opts) => {
           )
         ) {
           path.remove();
+          return;
         }
+
+        IMPLICITLY_ESM.add(state.file.ast.program);
```

Comment:
nit: Alternatively we can add this to the [`ExportDeclaration`](https://babeljs.io/docs/en/babel-types#exportdeclaration) visitor, it covers `ExportExportAllDeclaration`, `ExportDefaultDeclaration` and `ExportNamedDeclaration`.

Reply:
We would have to be careful about ordering, since we only want to add the node to the `WeakSet` if it's not removed.